### ISSUE TITLE
Color picker still sometimes fails

### ISSUE THREAD

**Issue by gallorob** [arayabrain/space-engineers-ai-spaceship-generator#33]
After clicking on the color picker enough times or after changing the viewed ship, the color will not be applied (sometimes). The log always shows `process_semaphore.is_locked=True` in such case.

**Comment by gallorob** [comment]
Bug fixed in https://github.com/arayabrain/space-engineers-ai-spaceship-generator/commit/d713f8c943a2e7a0982b384e3da85fbb275c4705

Leaving open until confirmed resolved.


_Note_:
This is not the best solution possible, but it is limited to the Dash framework. The better alternative would have been to check at the original interval if the `base_color` was different from the `color_picker.value` color and update accordingly, but that would have required updating the `content-plot.figure`, which is not possible since it's already an `Output` for `general_callback`.

**Comment by gallorob** [comment]
Apparently the color picking mini-window does not get disabled along the widget, that was annoying...

I changed the default behaviour of the widget in 692407fba7a862e5ed3d5ef7d2e13b28217c2ee3 to trigger the callback *only* when it loses focus. You should be able to now select whatever color you want, then clicking away will apply it to the spaceships. It's not _super_ intuitive, but at least you won't end up having a different color set in the widget from the spaceships.

Closing the issue for now since it should not even be possible to run into the same bug.

**Comment by martinpoliak-goodai** [comment]
Sorry, I still need to reopen this issue. I tested it and it works only partially. 
In Chrome, it works fine (color is applied after I click outside of the color picker) - OK.
In Firefox, the picked color is not applied until I close and reopen the color picker window - bug.

**Comment by gallorob** [comment]
That's not exactly true: Firefox opens a different kind of color picking widget (akin to old MS Paint), after you click "ok" on that window, just clicking on the page results in the color being applied. See here:

Firefox:
![colorpicking_firefox](https://user-images.githubusercontent.com/32876132/194437720-8e0cdb48-e9a4-4456-9602-b4fb5c18fd5e.gif)

Chrome/Edge:
![colorpicking_chrome](https://user-images.githubusercontent.com/32876132/194437949-aaa889a6-231e-4080-92a5-5621b9c0e440.gif)

So the behavior is the same-ish. Let me see if I can find a better configuration for this component that works well (and ideally, the same) across browsers.

**Comment by gallorob** [comment]
@martinpoliak-goodai it seems to be an issue with how Firefox differs in handling components focusing/defocusing due to the different widget.

I changed the layout to decouple color selection and application:
![image](https://user-images.githubusercontent.com/32876132/194443504-efc32f3b-efb5-4475-af4c-6626693f5516.png)
This way the color won't be applied until the user clicks the "Apply" button, and the behaviour is the same across browsers.

Closing the issue again as the bug is not present anymore.
